http-server/src/main/java/io/airlift/http/server/ReloadableSslContextFactoryProvider.java

```diff
@@ -12,69 +13,97 @@
         scheduledExecutor.scheduleWithFixedDelay(this::reload, refreshTime, refreshTime, MILLISECONDS);
     }
 
-    private static SslContextFactory.Server buildContextFactory(HttpServerConfig config)
+    private SslContextFactory.Server buildContextFactory()
     {
-        List<String> includedCipherSuites = config.getHttpsIncludedCipherSuites();
-        List<String> excludedCipherSuites = config.getHttpsExcludedCipherSuites();
-
         SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
-        Optional<KeyStore> pemKeyStore = tryLoadPemKeyStore(config);
+        Optional<KeyStore> pemKeyStore = tryLoadPemKeyStore(keystorePath, keystorePassword);
         if (pemKeyStore.isPresent()) {
             sslContextFactory.setKeyStore(pemKeyStore.get());
             sslContextFactory.setKeyStorePassword("");
         }
         else {
-            sslContextFactory.setKeyStorePath(config.getKeystorePath());
-            sslContextFactory.setKeyStorePassword(config.getKeystorePassword());
-            if (config.getKeyManagerPassword() != null) {
-                sslContextFactory.setKeyManagerPassword(config.getKeyManagerPassword());
+            sslContextFactory.setKeyStorePath(keystorePath);
+            sslContextFactory.setKeyStorePassword(keystorePassword);
+            if (keyManagerPassword != null) {
+                sslContextFactory.setKeyManagerPassword(keyManagerPassword);
             }
         }
-        if (config.getTrustStorePath() != null) {
-            Optional<KeyStore> pemTrustStore = tryLoadPemTrustStore(config);
+        if (trustStorePath != null) {
+            Optional<KeyStore> pemTrustStore = tryLoadPemTrustStore(trustStorePath);
             if (pemTrustStore.isPresent()) {
                 sslContextFactory.setTrustStore(pemTrustStore.get());
                 sslContextFactory.setTrustStorePassword("");
             }
             else {
-                sslContextFactory.setTrustStorePath(config.getTrustStorePath());
-                sslContextFactory.setTrustStorePassword(config.getTrustStorePassword());
+                sslContextFactory.setTrustStorePath(trustStorePath);
+                sslContextFactory.setTrustStorePassword(trustStorePassword);
             }
         }
         sslContextFactory.setIncludeCipherSuites(includedCipherSuites.toArray(new String[0]));
         sslContextFactory.setExcludeCipherSuites(excludedCipherSuites.toArray(new String[0]));
-        sslContextFactory.setSecureRandomAlgorithm(config.getSecureRandomAlgorithm());
+        sslContextFactory.setSecureRandomAlgorithm(secureRandomAlgorithm);
         sslContextFactory.setWantClientAuth(true);
-        sslContextFactory.setSslSessionTimeout((int) config.getSslSessionTimeout().getValue(SECONDS));
-        sslContextFactory.setSslSessionCacheSize(config.getSslSessionCacheSize());
+        sslContextFactory.setSslSessionTimeout((int) sslSessionTimeout.getValue(SECONDS));
```

Comment:
Move this conversion to the constructor `sslSessionTimeoutSeconds` and use `toIntExact()`